plugins/dap/dap-ui.nix

```diff
@@ -0,0 +1,209 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.dap.extensions.dap-ui;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption =
+    helpers.mkNullOrOption
+    (
+      with types;
+        either int (numbers.between 0.0 1.0)
+    );
+
+  mkMappingOption =
+    helpers.defaultNullOpts.mkNullable
+    (
+      with types;
+        attrsOf (either str (listOf str))
```

Comment:
This seems to be better as a submodule, it seems it can either be:
`"close"` or `"expand"|"open"|"remove"|"edit"|"repl"|"toggle"` ([from the source](https://github.com/rcarriga/nvim-dap-ui/blob/c020f660b02772f9f3d11f599fefad3268628a9e/lua/dapui/config/init.lua#L71-L73))